src/main/java/uk/gov/companieshouse/pscdataapi/service/CompanyPscService.java

```diff
@@ -220,13 +232,57 @@ public IndividualBeneficialOwner getIndividualBeneficialOwnerPsc(
                         "Individual Beneficial Owner PSC document not found in Mongo with id"
                                 + notificationId);
             }
+        } catch (ResourceNotFoundException rnfe) {
+            throw rnfe;
         } catch (Exception ex) {
             logger.error(ex.getMessage(), DataMapHolder.getLogMap());
             throw new ResourceNotFoundException(HttpStatus.NOT_FOUND,
                     "Unexpected error occurred while fetching PSC document");
         }
     }
 
+    private boolean determineShowFullDob(String companyNumber, Boolean registerView,
+                                         PscDocument pscDocument) throws ResourceNotFoundException {
+        if (!registerView) {
+            return false;
+        } else {
+            Optional<MetricsApi> companyMetrics = companyMetricsApiService
+                    .getCompanyMetrics(companyNumber);
+
+            if (companyMetrics.isPresent()) {
+                try {
+                    String registerMovedTo = companyMetrics.get().getRegisters()
+                            .getPersonsWithSignificantControl().getRegisterMovedTo();
+                    if (registerMovedTo.equals("public-register")) {
+                        Boolean isCeased = pscDocument.getData().getCeased();
```

Comment:
This will throw a null pointer exception if getData() returns null. It shouldn't happen but sometimes there is bad data in live so probably worth putting a null check before setting isCeased and ceasedOn

Reply:
I think we won't get a NullPointer on the **isCeased** and **ceasedOn** because in **CompanyPscService.getIndividualPsc** we are doing a filter for **document.getData().getKind()** and that will throw a 404: "Unexpected error occurred while fetching PSC document". I think we can get some NullPointers for the metrics **movedToPublicRegister**, but those will be caught by the try/catch block and will raise a 404: "not-on-public-register".